client/resource_manager/client/client.go

```diff
@@ -133,15 +136,25 @@ func (c *ResourceGroupsController) Stop() error {
 }
 
 func (c *ResourceGroupsController) putResourceGroup(ctx context.Context, name string) (*groupCostController, error) {
+	// ref https://github.com/tikv/pd/issues/5955
+	c.createGroupsControllerLock.Lock()
+	defer c.createGroupsControllerLock.Unlock()
```

Comment:
Maybe there is no need to introduce a lock here, we can just use `LoadOrStore` when we try to store the group into `c.groupsController`? If it already exists, just do nothing.